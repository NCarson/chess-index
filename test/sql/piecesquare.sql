

select pieces('k6K/8/8/8/8/8/8/p6P/ w - -'::board);

select expected_or_fail_bool('Q>pa1'::piecesquare::text = 'Q>pa1', true);
select expected_or_fail_bool('Q/ph8'::piecesquare::text = 'Q/ph8', true);
select expected_or_fail_bool('Q-pa1'::piecesquare::text = 'Q-pa1', true);

select expected_or_fail_int(
    (select count(*) from (select  fen = piecesquares_to_board(pieces(fen), footer(fen)::text )p from position) as t where p=false)::int, 0);
