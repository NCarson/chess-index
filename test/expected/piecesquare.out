select pieces('k6K/8/8/8/8/8/8/p6P/ w - -'::board);
      pieces       
-------------------
 {ka8,Kh8,pa1,Ph1}
(1 row)

select expected_or_fail_bool('Q>pa1'::piecesquare::text = 'Q>pa1', true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('Q/ph8'::piecesquare::text = 'Q/ph8', true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('Q-pa1'::piecesquare::text = 'Q-pa1', true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_int(
    (select count(*) from (select  fen = piecesquares_to_board(pieces(fen), footer(fen)::text )p from position) as t where p=false)::int, 0);
 expected_or_fail_int 
----------------------
 
(1 row)

